frontend/src/pages/ManageData/CSVUpload.js

```diff
@@ -69,26 +120,96 @@ function CSVUpload(props) {
           );
         })}
       </Select>
-      <br />
+      <p>School</p>
       <Select
         label="School"
         variant="outlined"
         placeholder="School"
         required="true"
         name="selectedSchool"
+        displayEmpty
         value={field.selectedSchool}
         onChange={updateState}
+        disabled={field.selectedDistrict === ''}
       >
-        {schoolSet.map(schoolData => {
-          return (
-            <MenuItem value={schoolData.schoolId}>
-              {schoolData.schoolName}
-            </MenuItem>
-          );
+        <MenuItem value="" disabled>
+          Select one
+        </MenuItem>
+        {schools.map(school => {
+          if (school.districtId === field.selectedDistrict)
+            return (
+              <MenuItem value={school.schoolId}>{school.schoolName}</MenuItem>
+            );
         })}
       </Select>
+      <p>Final</p>
+      <Select
+        label="Final"
+        variant="outlined"
+        placeholder="Final"
+        required="true"
+        name="selectedFinal"
+        value={field.selectedFinal}
+        onChange={updateState}
+        disabled={field.selectedSchool === ''}
+      >
+        <MenuItem value>Final</MenuItem>
+        <MenuItem value={false}>Not Final</MenuItem>
+      </Select>
+      <p>File</p>  
+      <input
+        accept="*"
+        style={{display: "none"}}
+        id="outlined-button-file"
+        multiple
+        type="file"
+        name="csvFileName"
+        required="true"
+        label="File"
+        onChange={updateState}
+      />
+      <label htmlFor="outlined-button-file">
+        <Button variant="outlined" component="span" style={{margin: "theme.spacing(1)"}}>
+          Browse
+        </Button>
+      </label>
+     
```

Comment:
I would probably try and add some feedback here when the file is uploaded, otherwise it looks like nothing happened.  Something simple like ```<p>{field.csv && `Uploaded file: ${field.csv.name}`}</p>``` could work for now